beacon-chain/powchain/log_processing.go

```diff
@@ -259,12 +260,42 @@ func (s *Service) processPastLogs(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	// to store all blocks
+	headersMap := make(map[uint64]*gethTypes.Header)
+
+	// batch request the desired headers and store them in a
+	// map for quick access
+	requestHeaders := func(startBlk uint64, endBlk uint64) error {
+		headers, err := s.batchRequestHeaders(startBlk, endBlk)
+		if err != nil {
+			return err
+		}
+		for _, h := range headers {
+			if h != nil && h.Number != nil {
+				headersMap[h.Number.Uint64()] = h
```

Comment:
Do we not need mutex protection for `headersMap`?

Reply:
this is all single-threaded, so using a mutex isnt required since there isnt any concurrent access

Reply:
Cool. What i thought but still wanted to confirm